src/app.js

```diff
@@ -0,0 +1,369 @@
+import { encode, decode } from './encode';
+
+const DOC_URL = 'https://github.com/matteo-ronchetti/aws-lambda-power-tuning-ui';
+const HASH_DELIM = ';';
+const NORMAL_HASH_LEN = 3;
+const COMPARISON_HASH_LEN = 8;
+
+const chartColors = {
+  red: 'rgb(255, 99, 132)',
+  blue: 'rgb(54, 162, 235)',
+  orange: 'rgb(255, 165, 00)',
+  green: 'rgb(34, 139, 34)',
+};
+
+const chartOptions = {
+  responsive: true,
+  hoverMode: 'index',
+  tooltips: {
+    mode: 'index',
+    intersect: false,
+    enabled: false,
+    custom: customTooltip,
+  },
+  stacked: false,
+  title: {
+    display: false,
+  },
+  scales: {
+    xAxes: [
+      {
+        scaleLabel: {
+          display: true,
+          labelString: 'Memory / Power (MB)',
+        },
+      },
+    ],
+    yAxes: [
+      {
+        ticks: {
+          beginAtZero: true,
+          callback: (value /*, index, values*/) => `${smartRound(value)} ms`,
+        },
+        type: 'linear', // only linear but allow scale type registration. This allows extensions to exist solely for log scale for instance
+        display: true,
+        scaleLabel: {
+          display: true,
+          labelString: 'Invocation Time',
+        },
+        position: 'left',
+        id: 'time-axis',
+      },
+      {
+        ticks: {
+          beginAtZero: true,
+          callback: (value /*, index, values*/) => `${smartRound(value)} $`,
+        },
+        type: 'linear', // only linear but allow scale type registration. This allows extensions to exist solely for log scale for instance
+        display: true,
+        position: 'right',
+        id: 'cost-axis',
+        scaleLabel: {
+          display: true,
+          labelString: 'Invocation Cost',
+        },
+        // grid line settings
+        gridLines: {
+          drawOnChartArea: false, // only want the grid lines for one axis to show up
+        },
+      },
+    ],
+  },
+};
+
+export class App {
+  constructor(store) {
+    this.store = store;
+    this.canvas = document.querySelector('canvas');
+    this.reportElem = document.getElementById('report');
+    this.errorElem = document.getElementById('error');
+
+    // App State
+    this.data1 = null;
+    this.data1URL = '';
+    this.data2 = null;
+    this.data2URL = '';
+    this.legend1 = '';
+    this.legend2 = '';
+
+    this.store.addListener(this.show.bind(this), this.error.bind(this));
+  }
+
+  setData1(parts) {
```

Comment:
Should these setData methods also set `this.legend1` and `this.legend2`?

We'd avoid these ([here](https://github.com/matteo-ronchetti/aws-lambda-power-tuning-ui/pull/14/files#diff-c72a907ac323cd2f334ed0e2bd07d15ab62581c4753660c8a0d1c681b30be4b6R164-R165)):
```js
this.legend1 = decodeURIComponent(hashParts[6]);
this.legend2 = decodeURIComponent(hashParts[7]);
```

Reply:
We could combine it, but `setData2` would no longer be a suitable name for what it is doing. I think it is clearer the way it is.